src/renderer/components/badge/badge.tsx

```diff
@@ -40,12 +40,26 @@ export const Badge = withTooltip(observer(({
   children,
   ...elemProps
 }: BadgeProps) => {
-  const elem = createRef<HTMLDivElement>();
+  const elem = useRef<HTMLDivElement>(null);
+  const elemHasBeenSet = useRef(false);
   const [isExpanded, setIsExpanded] = useState(false);
+  const [isExpandable, setIsExpandable] = useState(expandable);
 
-  const isExpandable = expandable && elem.current
-    ? elem.current.clientWidth < elem.current.scrollWidth
-    : false;
+  useEffect(() => {
+    if (elemHasBeenSet.current) {
+      return;
+    }
+
+    if (elem.current) {
+      elemHasBeenSet.current = true;
+    }
```

Comment:
This check is here because we only want to compute if a badge is expandable once (when it is closed). Otherwise it will be re-computed when it is clicked and then expanded.